### PR TITLE
Do rms_resource_free() in rso_prepare_to_stop()

### PR DESCRIPTION
**Pull request by huanghua78**
* In commit 3af2fc786e, we do reverse session cleanup in prepare_to_stop().
So, we must also do rms_resource_free() in rso_prepare_to_stop().
This patch fixes the rm-ut:rmvsv UT failure:

Thread 29 (Thread 0x7f77553f4700 (LWP 1763408)):
0  0x00007f77adb9ada8 in nanosleep () from /lib64/libc.so.6
1  0x00007f77adb9acae in sleep () from /lib64/libc.so.6
2  0x00007f77afc34af2 in m0_debugger_invoke () at lib/user_space/uassert.c:144
3  0x00007f77afc34c04 in m0_arch_panic (c=c@entry=0x7f77b00fa700 <signal_panic>, ap=ap@entry=0x7f77553f2538) at lib/user_space/uassert.c:129
4  0x00007f77afc22ba7 in m0_panic (ctx=ctx@entry=0x7f77b00fa700 <signal_panic>) at lib/assert.c:52
5  0x00007f77afc34c48 in sigsegv (sig=11) at lib/user_space/ucookie.c:52
6  <signal handler called>
7  m0_rpc_conn_is_snd (conn=0x5f5f5f5f5f5f5f5f) at rpc/conn.c:282
8  0x00007f77afcbe0a1 in m0_rpc_post (item=item@entry=0x7f773d820130) at /home/520428/work/cortx-motr/rpc/rpc_internal.h:97
9  0x00007f77b136c498 in m0_rm_outgoing_send (outgoing=outgoing@entry=0x7f773d820050) at /home/520428/work/cortx-motr/rm/rm_fops.c:310
10 0x00007f77b136c7dd in outgoing_queue (other=0x7f77a419ce50, in=0x7f773f0e63d0, outreq=0x7f773d820050, owner=0x7f77a41ca0d0, otype=M0_ROT_REVOKE)
   at /home/520428/work/cortx-motr/rm/rm_fops.c:351
11 m0_rm_request_out (otype=otype@entry=M0_ROT_REVOKE, in=in@entry=0x7f773f0e63d0, loan=loan@entry=0x7f77a419d000, credit=credit@entry=0x7f77553f3750,
   other=other@entry=0x7f77a419ce50) at /home/520428/work/cortx-motr/rm/rm_fops.c:392
12 0x00007f77afca65df in revoke_send (credit=0x7f77a419d000, loan=0x7f77a419d000, in=0x7f773f0e63d0) at rm/rm.c:2633
13 incoming_check_with (rest=0x7f77553f36f0, in=0x7f773f0e63d0) at rm/rm.c:2437
14 incoming_check (in=0x7f773f0e63d0) at rm/rm.c:2060
15 owner_balance (o=o@entry=0x7f77a41ca0d0) at rm/rm.c:1995
16 0x00007f77afca6cdc in owner_liquidate (o=0x7f77a41ca0d0) at rm/rm.c:864
17 owner_finalisation_check (owner=0x7f77a41ca0d0) at rm/rm.c:551
18 owner_balance (o=o@entry=0x7f77a41ca0d0) at rm/rm.c:2002
19 0x00007f77afca6eb4 in owner_windup_locked (owner=owner@entry=0x7f77a41ca0d0) at rm/rm.c:927
20 0x00007f77afca6f38 in m0_rm_owner_windup (owner=owner@entry=0x7f77a41ca0d0) at rm/rm.c:936
21 0x00007f77afcaa309 in rms_resources_free (rtype=rtype@entry=0x7f773f0ed370) at rm/rm_service.c:183
22 0x00007f77afcaa3ee in rms_stop (service=0x7f773f0ed030) at rm/rm_service.c:207
23 0x00007f77afc9da6e in m0_reqh_service_stop (service=service@entry=0x7f773f0ed030) at reqh/reqh_service.c:419
24 0x00007f77b12f2fab in cs_service_fini (service=0x7f773f0ed030) at /home/520428/work/cortx-motr/motr/setup.c:1190
25 0x00007f77b12f31d0 in reqh_context_services_fini (cctx=0x7f77b183e1a8 <sctx+40>, rctx=0x7f77b183e4b8 <sctx+824>) at /home/520428/work/cortx-motr/motr/setup.c:1216
26 cs_level_leave (module=<optimized out>) at /home/520428/work/cortx-motr/motr/setup.c:2849
27 0x00007f77afc757d0 in m0_module_fini (module=module@entry=0x7f77b184cc00 <sctx+60032>, level=level@entry=-1) at module/module.c:154
28 0x00007f77b12f6fcd in m0_cs_fini (cctx=cctx@entry=0x7f77b183e1a8 <sctx+40>) at /home/520428/work/cortx-motr/motr/setup.c:3056
29 0x00007f77afcc1009 in m0_rpc_server_stop (sctx=sctx@entry=0x7f77b183e180 <sctx>) at rpc/rpclib.c:89
30 0x00007f77b136f4fa in rm_service_stop (sctx=0x7f77b183e180 <sctx>) at rm/ut/rm_service.c:84
31 rm_svc_server (tid=<optimized out>) at rm/ut/rm_service.c:84
32 0x00007f77afc29f7a in m0_thread_trampoline (arg=arg@entry=0x7f77b5e92008 <rm_ctxs+8>) at lib/thread.c:117
33 0x00007f77afc357bd in uthread_trampoline (arg=0x7f77b5e92008 <rm_ctxs+8>) at lib/user_space/uthread.c:98
34 0x00007f77af2c915a in start_thread () from /lib64/libpthread.so.0
35 0x00007f77adbcedd3 in clone () from /lib64/libc.so.6

* Fix UT ha-state-ut:ha-poolversion-get failure.
Actual pver is returned in case when nothing better (cleaner) can be found.
This new mechanism is introduced in 2f43b29b6ce7998b61f298910f188ddce0babf3a .

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
<h1>Jenkins CI Result : <a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454/>Motr#1454</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>2</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/49motr-rpc-cancel target=_blank>04motr-single-node/49motr-rpc-cancel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>32</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/28sys-kvs target=_blank>01motr-single-node/28sys-kvs</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/04initscripts target=_blank>01motr-single-node/04initscripts</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/20rpc-session-cancel target=_blank>02motr-single-node/20rpc-session-cancel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/10pver-assign target=_blank>02motr-single-node/10pver-assign</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/21fsync-single-node target=_blank>02motr-single-node/21fsync-single-node</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/13dgmode-io target=_blank>02motr-single-node/13dgmode-io</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/14poolmach target=_blank>02motr-single-node/14poolmach</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/11m0t1fs target=_blank>02motr-single-node/11m0t1fs</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/26motr-user-kernel-tests target=_blank>02motr-single-node/26motr-user-kernel-tests</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/08spiel target=_blank>02motr-single-node/08spiel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/06conf target=_blank>03motr-single-node/06conf</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/34sns-repair-1n-1f target=_blank>04motr-single-node/34sns-repair-1n-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/08spiel-sns-repair-quiesce target=_blank>04motr-single-node/08spiel-sns-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/28sys-kvs-kernel target=_blank>04motr-single-node/28sys-kvs-kernel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/11m0t1fs-rconfc-fail target=_blank>04motr-single-node/11m0t1fs-rconfc-fail</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/08spiel-sns-repair target=_blank>04motr-single-node/08spiel-sns-repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/19sns-repair-abort target=_blank>04motr-single-node/19sns-repair-abort</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/22sns-repair-ios-fail target=_blank>04motr-single-node/22sns-repair-ios-fail</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/18sns-repair-quiesce target=_blank>05motr-single-node/18sns-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/12fwait target=_blank>05motr-single-node/12fwait</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/16sns-repair-multi target=_blank>05motr-single-node/16sns-repair-multi</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/07mount-fail target=_blank>05motr-single-node/07mount-fail</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/15sns-repair-single target=_blank>05motr-single-node/15sns-repair-single</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/23sns-abort-quiesce target=_blank>05motr-single-node/23sns-abort-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/17sns-repair-concurrent-io target=_blank>05motr-single-node/17sns-repair-concurrent-io</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/07mount target=_blank>05motr-single-node/07mount</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/07mount-multiple target=_blank>05motr-single-node/07mount-multiple</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/12fsync target=_blank>05motr-single-node/12fsync</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>41</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/43m0crate target=_blank>01motr-single-node/43m0crate</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/05confgen target=_blank>01motr-single-node/05confgen</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/06hagen target=_blank>01motr-single-node/06hagen</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/52motr-singlenode-sanity target=_blank>01motr-single-node/52motr-singlenode-sanity</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/01net target=_blank>01motr-single-node/01net</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/03console target=_blank>01motr-single-node/03console</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/01motr-single-node/02rpcping target=_blank>01motr-single-node/02rpcping</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/07m0d-fatal target=_blank>02motr-single-node/07m0d-fatal</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/67fdmi-plugin-multi-filters target=_blank>02motr-single-node/67fdmi-plugin-multi-filters</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/61sns-repair-motr-1n-1f target=_blank>03motr-single-node/61sns-repair-motr-1n-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/72spiel-sns-motr-repair-quiesce target=_blank>03motr-single-node/72spiel-sns-motr-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/08spiel-multi-confd target=_blank>03motr-single-node/08spiel-multi-confd</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/69sns-repair-motr-quiesce target=_blank>03motr-single-node/69sns-repair-motr-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/62sns-repair-motr-mf target=_blank>03motr-single-node/62sns-repair-motr-mf</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/70sns-failure-after-repair-quiesce target=_blank>03motr-single-node/70sns-failure-after-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/63sns-repair-motr-1k-1f target=_blank>03motr-single-node/63sns-repair-motr-1k-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/60sns-repair-motr-1f target=_blank>03motr-single-node/60sns-repair-motr-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/66sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/66sns-repair-motr-abort-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/24motr-dix-repair-lookup-insert-spiel target=_blank>03motr-single-node/24motr-dix-repair-lookup-insert-spiel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/68sns-repair-motr-shutdown target=_blank>03motr-single-node/68sns-repair-motr-shutdown</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/64sns-repair-motr-ios-fail target=_blank>03motr-single-node/64sns-repair-motr-ios-fail</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/71spiel-sns-motr-repair target=_blank>03motr-single-node/71spiel-sns-motr-repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/24motr-dix-repair-lookup-insert-m0repair target=_blank>03motr-single-node/24motr-dix-repair-lookup-insert-m0repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/04sss target=_blank>03motr-single-node/04sss</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/03motr-single-node/65sns-repair-motr-abort target=_blank>03motr-single-node/65sns-repair-motr-abort</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/48motr-raid0-io target=_blank>04motr-single-node/48motr-raid0-io</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/25m0kv target=_blank>04motr-single-node/25m0kv</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/44motr-rm-lock-cc-io target=_blank>04motr-single-node/44motr-rm-lock-cc-io</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/04motr-single-node/45motr-rmw target=_blank>04motr-single-node/45motr-rmw</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/23dix-repair-m0repair target=_blank>05motr-single-node/23dix-repair-m0repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/43motr-sync-replication target=_blank>05motr-single-node/43motr-sync-replication</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/42motr-utils target=_blank>05motr-single-node/42motr-utils</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/45motr-sns-repair-N-1 target=_blank>05motr-single-node/45motr-sns-repair-N-1</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/40motr-dgmode target=_blank>05motr-single-node/40motr-dgmode</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/23dix-repair-quiesce-m0repair target=_blank>05motr-single-node/23dix-repair-quiesce-m0repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/23spiel-dix-repair-quiesce target=_blank>05motr-single-node/23spiel-dix-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/44motr-sns-repair target=_blank>05motr-single-node/44motr-sns-repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit/05motr-single-node/23spiel-dix-repair target=_blank>05motr-single-node/23spiel-dix-repair</a></p></details></td></tr><tr><td>Total</td><td>75</td><td><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1454//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

**Comment by huanghua78** [comment]
I am sorry, but why is the lnet test included in the UT?
lnet is not added in the transport list as default.
@madhavemuri @upendrapatwardhan

**Comment by upendrapatwardhan** [comment]
> I am sorry, but why is the lnet test included in the UT? lnet is not added in the transport list as default. @madhavemuri @upendrapatwardhan

Please see the below comments.
https://github.com/Seagate/cortx-motr/pull/1799#discussion_r882762613

https://github.com/Seagate/cortx-motr/pull/1809#discussion_r882754083

Based on above inputs, even if lnet is not the default transport, the lnet UT is enabled if the lustre rpms are installed and m0tr.ko is available.

**Comment by huanghua78** [comment]
Hello @andriytk , can you please help check this failure? https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1463//testReport/junit/01motr-single-node/00userspace-tests

I verified this UT failure locally.
It was introduced recently.

**Review by madhavemuri — APPROVED** [review]

**Comment by huanghua78** [comment]
> 



> Hello @andriytk , can you please help check this failure? https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1463//testReport/junit/01motr-single-node/00userspace-tests
> 
> I verified this UT failure locally. It was introduced recently in [2f43b29](https://github.com/Seagate/cortx-motr/commit/2f43b29b6ce7998b61f298910f188ddce0babf3a)


And this is the fix:
```
diff --git a/ha/ut/note.c b/ha/ut/note.c
index 531a055e6..7fc57a474 100644
--- a/ha/ut/note.c
+++ b/ha/ut/note.c
@@ -493,6 +493,7 @@ static void test_poolversion_get(void)
        struct m0_conf_pver    *pver2 = NULL;
        struct m0_conf_pver    *pver3 = NULL;
        struct m0_conf_pver    *pver4 = NULL;
+       struct m0_conf_pver    *pver5 = NULL;
        struct m0_reqh          reqh;
        struct m0_confc        *confc = m0_reqh2confc(&reqh);
        int                     rc;
@@ -542,10 +543,10 @@ static void test_poolversion_get(void)
        M0_UT_ASSERT(recd_disks(pver0) == 2);

        ha_state_accept_1(&disk_76, M0_NC_TRANSIENT);
-       rc = m0_conf_pver_get(confc, &pool4_fid, &pver3);
-       M0_UT_ASSERT(rc == -ENOENT);
-       M0_UT_ASSERT(pver3 == NULL);
-
+       rc = m0_conf_pver_get(confc, &pool4_fid, &pver5);
+       M0_UT_ASSERT(rc == 0);
+       M0_UT_ASSERT(pver5 == pver0);
+       M0_UT_ASSERT(pver5->pv_kind == M0_CONF_PVER_ACTUAL);

        rc = m0_conf_pver_get(confc, &pool56_fid, &pver3);
        M0_UT_ASSERT(rc == 0);
@@ -574,6 +575,7 @@ static void test_poolversion_get(void)
        m0_confc_close(&pver2->pv_obj);
        m0_confc_close(&pver3->pv_obj);
        m0_confc_close(&pver4->pv_obj);
+       m0_confc_close(&pver5->pv_obj);
        ha_ut_conf_fini(&reqh);
        m0_reqh_fini(&reqh);
 }
```

@andriytk , please verify.

File: rm/rm_service.c [hunk 1]
```diff
@@ -62,6 +63,7 @@ static const struct m0_reqh_service_type_ops rms_type_ops = {
 static const struct m0_reqh_service_ops rms_ops = {
 	.rso_start           = rms_start,
 	.rso_start_async     = m0_reqh_service_async_start_simple,
+	.rso_prepare_to_stop = rms_prepare_to_stop,
```
Comment (andriytk) [inline rm/rm_service.c:66]: How/where this new callback is used?

**Review by andriytk — APPROVED** [review]

**Pull request merged**